src/media_tools/av_parsers.c

```diff
@@ -4385,6 +4385,133 @@ GF_Err gf_media_prores_parse_bs(GF_BitStream *bs, GF_ProResFrameInfo *prores_fra
 
+GF_EXPORT
+const char *gf_iamf_get_obu_name(IamfObuType obu_type)
+{
+        switch (obu_type) {
+                case OBU_IA_CODEC_CONFIG: return "codec_config";
+                case OBU_IA_AUDIO_ELEMENT: return "audio_element";
+                case OBU_IA_MIX_PRESENTATION: return "mix_presentation";
+                case OBU_IA_PARAMETER_BLOCK: return "parameter_block";
+                case OBU_IA_TEMPORAL_DELIMITER: return "temporal_delimiter";
+                case OBU_IA_AUDIO_FRAME: return "audio_frame";
+                case OBU_IA_AUDIO_FRAME_ID0: return "audio_frame_id0";
+                case OBU_IA_AUDIO_FRAME_ID1: return "audio_frame_id1";
+                case OBU_IA_AUDIO_FRAME_ID2: return "audio_frame_id2";
+                case OBU_IA_AUDIO_FRAME_ID3: return "audio_frame_id3";
+                case OBU_IA_AUDIO_FRAME_ID4: return "audio_frame_id4";
+                case OBU_IA_AUDIO_FRAME_ID5: return "audio_frame_id5";
+                case OBU_IA_AUDIO_FRAME_ID6: return "audio_frame_id6";
+                case OBU_IA_AUDIO_FRAME_ID7: return "audio_frame_id7";
+                case OBU_IA_AUDIO_FRAME_ID8: return "audio_frame_id8";
+                case OBU_IA_AUDIO_FRAME_ID9: return "audio_frame_id9";
+                case OBU_IA_AUDIO_FRAME_ID10: return "audio_frame_id10";
+                case OBU_IA_AUDIO_FRAME_ID11: return "audio_frame_id11";
+                case OBU_IA_AUDIO_FRAME_ID12: return "audio_frame_id12";
+                case OBU_IA_AUDIO_FRAME_ID13: return "audio_frame_id13";
+                case OBU_IA_AUDIO_FRAME_ID14: return "audio_frame_id14";
+                case OBU_IA_AUDIO_FRAME_ID15: return "audio_frame_id15";
+                case OBU_IA_AUDIO_FRAME_ID16: return "audio_frame_id16";
+                case OBU_IA_AUDIO_FRAME_ID17: return "audio_frame_id17";
+                case OBU_IA_RESERVED_24:
+                case OBU_IA_RESERVED_25:
+                case OBU_IA_RESERVED_26:
+                case OBU_IA_RESERVED_27:
+                case OBU_IA_RESERVED_28:
+                case OBU_IA_RESERVED_29:
+                case OBU_IA_RESERVED_30:
+                        return "reserved";
+                case OBU_IA_SEQUENCE_HEADER: return "ia_sequence_header";
+                default: return "unknown";
+        }
+}
+
+#ifndef GPAC_DISABLE_AV_PARSERS
+
+GF_EXPORT
+GF_Err gf_iamf_parse_obu(GF_BitStream *bs, IamfObuType *obu_type, u64 *obu_size)
+{
+        GF_Err e = GF_OK;
+        u64 pos = gf_bs_get_position(bs);
+
+        if (!bs || !obu_type) {
+              return GF_BAD_PARAM;
+        }
+
+        gf_bs_mark_overflow(bs, GF_TRUE);
+
+        e = gf_iamf_parse_obu_header(bs, obu_type, obu_size);
+
+        if (gf_bs_is_overflow(bs)) {
+              GF_LOG(GF_LOG_WARNING, GF_LOG_CODING, ("[IAMF] OBU parsing consumed too many bytes.\n"));
+              e = GF_NON_COMPLIANT_BITSTREAM;
+        }
+        gf_bs_seek(bs, pos + *obu_size);
+        return e;
+}
+
+GF_EXPORT
```

Comment:
I think this function could be static (and removed from the header). Any reason to export it?

Reply:
you're right, changed to be static and removed from header